### DISCUSSION TITLE
TPI3 Eventos repetitivos

### DISCUSSION THREAD

**Discussion by ivanscardaoni1** [dds-jv/foro#292] [Consultas de objetos] [answered]
Buenas! Tengo una duda respecto a los eventos repetitivos.

Lo que vimos en clase respecto a la repetición de eventos fue que un determinado evento se repita cada cierto tiempo, pero en este caso tenemos que repetir una instancia de un evento y que esta pueda verse en el calendario, no debería dispararse ese evento cada determinado tiempo, sino que ya debería existir. El problema es que en el enunciado dice "no deseamos precalcular y precargar cada instancia del evento como un evento separado" ¿Que significa esta frase? Por otro lado, si tuviese que precalcular las fechas y ubicar la misma instancia del evento en cada fecha ¿Que pasa si el evento se repite infinitamente? ¿Seria un método que jamas terminaría de ejecutar?

Sinceramente estoy un poco perdido con este requerimiento.

Desde ya muchas gracias!

**Comment by gastonprieto** [comment, marked as answer]
Buenas @ivanscardaoni1 

No podría estar muy seguro de lo que vieron en clase, esa me la perdi :P

> El problema es que en el enunciado dice "no deseamos precalcular y precargar cada instancia del evento como un evento separado" ¿Que significa esta frase? Por otro lado, si tuviese que precalcular las fechas y ubicar la misma instancia del evento en cada fecha ¿Que pasa si el evento se repite infinitamente? ¿Seria un método que jamas terminaría de ejecutar?

Esto se refiere a que si tienen un evento diario, que dura una semana. Entonces *no* carguen 7 eventos simples en el calendario, sino que debería estar representado por un único evento que actue como los 7.

**Reply by ivanscardaoni1** [reply]
Perfecto, con eso se entiende mejor. Interpreto que la idea seria que ese único evento, cuando uno le pide las ocurrencias entre una fecha y otra, las calcule basándose en su inicio, frecuencia y fin, pero se CALCULA cuando se pide, no se INSTANCIA.
Comprendo bien?

**Reply by gastonprieto** [reply]
en tu calculo tal vez instancias objetos y los retornas.

Pero no los tenes pre instanciandos, antes de que se los pidan.
